tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/UpdateCombinationStockFeatureContext.php

```diff
@@ -0,0 +1,153 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain\Product\Combination;
+
+use Behat\Gherkin\Node\TableNode;
+use DateTime;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\UpdateCombinationStockCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CombinationConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationStock;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime as DateTimeUtil;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+
+class UpdateCombinationStockFeatureContext extends AbstractCombinationFeatureContext
+{
+    /**
+     * @When I update combination :combinationReference stock with following details:
+     *
+     * @param string $combinationReference
+     * @param TableNode $tableNode
+     */
+    public function updateStock(string $combinationReference, TableNode $tableNode): void
+    {
+        $command = new UpdateCombinationStockCommand($this->getSharedStorage()->get($combinationReference));
+        $this->fillCommand($command, $tableNode->getRowsHash());
+
+        $this->getCommandBus()->handle($command);
+    }
+
+    /**
+     * @Transform table:combination stock detail,value
+     *
+     * @param TableNode $tableNode
+     *
+     * @return CombinationStock
+     */
+    public function transformCombinationStock(TableNode $tableNode): CombinationStock
+    {
+        $dataRows = $tableNode->getRowsHash();
+
+        return new CombinationStock(
+            (int) $dataRows['quantity'],
+            (int) $dataRows['minimal quantity'],
+            (int) $dataRows['low stock threshold'],
+            PrimitiveUtils::castStringBooleanIntoBoolean($dataRows['low stock alert is on']),
+            $dataRows['location'],
+            DateTimeUtil::NULL_DATE === $dataRows['available date'] ? null : new DateTime($dataRows['available date'])
```

Comment:
It would be easier, I think, to rely on empty string for this `null` value instead of using `0000:00:00` inside the behat scenarios